src/DataReader/parse_queries.rs

```diff
@@ -172,6 +175,56 @@ fn build_expression_from_pair(pair: pest::iterators::Pair<Rule>) -> QueryExpress
     }
 }
 
+fn build_state_from_pair(pair: pest::iterators::Pair<Rule>) -> QueryExpression {
+    let mut inner_pair = pair.clone().into_inner();
+    let locPair = inner_pair.next().unwrap();
+
+    let mut loc_names: Vec<Box<QueryExpression>> = Vec::new();
+    for loc_name in locPair.as_str().split(',') {
+        loc_names.push(Box::new(QueryExpression::LocName(
+            loc_name.trim().to_string(),
+        )));
+    }
+
+    let clockPair = inner_pair.next().unwrap();
+
+    // In the following line of code, we build a BoolExprssion based on the clock constraints defined for the given location.
+    // To make BoolExprssion we use the InvariantParser parser instead.
+    // Becuase clocks is defined as c1&&c2... in the InvariantParser we replace ',' to match the format e.g., e.g., "x>0,y<5" => "x>0&&y<5"
+    let invariantVersion: Option<Box<BoolExpression>> = if clockPair.as_str().trim() != "" {
+        let clockString = clockPair.as_str().trim().to_string().replace(',', "&&");
+        let invariantVersion = match parse(&clockString) {
+            Ok(result) => result,
+            Err(result) => panic!("{:?}", result),
+        };
+        Some(Box::new(invariantVersion))
+    } else {
+        None
+    };
+
+    match pair.as_rule() {
+        Rule::state => QueryExpression::State(loc_names, invariantVersion),
+        err => panic!("Unable to match: {:?} as rule loc or clocks", err),
+    }
```

Comment:
Same as above

Reply:
Pair does not implement expect(), so we a bit unsure how to simplify it.

Reply:
My bad, didn't notice :)